protocol/0053-PERP-product_builtin_perpetual_future.md

```diff
@@ -4,12 +4,16 @@ This built-in product provides perpetual futures that are cash-settled, i.e. the
 
 [Background reading](https://www.paradigm.xyz/2021/05/everlasting-options/#Perpetual_Futures)
 
-Perpetual futures are a simple "delta one" product.
+Perpetual futures are a simple "delta one" product. Mark-to-market settlement occurs with a predefined frequency as per [0003-MTMK-mark_to_market_settlement](0003-MTMK-mark_to_market_settlement.md).  Additionally, a settlement using external data is carried out whenever `settlement_cue` is triggered AND the `settlement_data` is received. A number of protective measures can be specified for a market to deal with data scarcity in a predefined way.
 
 ## 1. Product parameters
 
-1. `settlement_data (Data Source: number)`: this data is used by the product to calculate periodic settlement cashflows. The receipt of this data triggers this calculation and the transfers between parties to "true up" to the external reference price.
 1. `settlement_asset (Settlement Asset)`: this is used to specify the single asset that an instrument using this product settles in.
+1. `settlement_cue (Data Source)`: this data is used to indicate that next periodic settlement should happen imminently.
+1. `settlement_data (Data Source: number)`: this data is used by the product to calculate periodic settlement cashflows. The receipt of this data triggers this calculation and the transfers between parties to "true up" to the external reference price.
+1. `settlement_cue_auction_duration`: a time interval which specifies the duration of an auction started once settlement cue is received. The auction ends when the specified time elapses or when the settlement data is received. A value of `0s` indicates no auction.
+1. `max_settlement_gap`: a time interval which specifies the amount of time without periodic settlement after which the market will go into protective auction and remain in that mode until settlement data is received.
```

Comment:
Do we also want/need a `min_settlement_gap`? (Might be somewhat confusing if you're expecting settlement once a day and someone sends a settlement half way through. But perhaps that's not an issue?)

Reply:
I was thinking the `settlement_data` oracle will handle that (we still need to spec it of course), but it could have a mechanism that fires up the data on same dates as `settlement_cue` subject to data meeting some criteria. 

`settlement_cue` will probably just be an internal oracle that fires with a specified frequency in most cases so in a sense it will already contain the min gap.

Reply:
Ah yes makes sense :+1: